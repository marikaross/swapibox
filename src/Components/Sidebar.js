import React from 'react';
import PropTypes from 'prop-types';

const Sidebar = ({ summary }) => {
  return (
    <div>
      <section>
     {/*   <h3>{summary}</h3>
        <h3>{summary}</h3>
        <h3>{summary}*/}
      </section>
    </div>
    )
}

export default Sidebar;