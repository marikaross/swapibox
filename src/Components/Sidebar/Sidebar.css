.sidebar {
  width: 30%;
  color: #56C0D4;
  background-color: black;
  overflow: hidden; }

.scrollContainer {
  display: flex;
  justify-content: center;
  height: 800px;
  perspective: 400px;
  font-weight: 600;
  letter-spacing: 4px;
  line-height: 150%;
  text-align: justify;
  margin: 20px; }

.scroll {
  top: 9999px;
  position: relative;
  transform-origin: 50% 100%;
  animation: crawl 60s linear; }

@keyframes scroll {
  0% {
    top: 0;
    transform: rotateX(20deg) translateZ(0); }
  100% {
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px); } }
