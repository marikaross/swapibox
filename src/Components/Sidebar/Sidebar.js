import React from 'react';
import PropTypes from 'prop-types';

const Sidebar = ({ summary }) => {
  return (
    <div className="sidebar">
      <img src={require ('../../images/starWarsLogo.jpg')} />
      <section className="scroll">
        <h3>{summary.summary}</h3>
        <h3>{summary.title}</h3>
        <h3>{summary.releaseDate}</h3>
      </section>
    </div>
    )
}

export default Sidebar;


Sidebar.propTypes = {
  summary: PropTypes.object.isRequired,
};