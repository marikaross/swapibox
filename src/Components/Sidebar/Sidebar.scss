.sidebar {
  width: 30%;
  color: #56C0D4;
  background-color: black;
  overflow: hidden;
}

.scrollContainer {
  justify-content: center;
  width: 90%;
  margin: 20px;
  font-weight: 600;
  letter-spacing: 4px;
  line-height: 150%;
  text-align: justify;
}

.scroll {
  position: absolute;
  animation: scroll 100s linear 4s infinite;
  width: 25%;
}


@keyframes scroll {
    0% {
      top: 100%;
    }
    100% {
      top: 0%;
    }
}