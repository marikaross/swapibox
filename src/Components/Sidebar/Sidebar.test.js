import React from 'react';
import { shallow } from 'enzyme';
import Sidebar from './Sidebar.js';

describe('Sidebar', () => {
  it.only('should match the snapshot with all data passed in correctly', () => {
    const mockScroll = {summary: 'This is a fake scroll', 
      title: 'There\'s no hope', 
      releaseDate: '2018-07-07', 
     }

    

    const wrapper = shallow(<Sidebar summary={mockScroll} />)
    expect(wrapper).toMatchSnapshot();

  })
})