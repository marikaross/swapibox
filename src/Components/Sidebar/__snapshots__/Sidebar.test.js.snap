// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar
    summary={
        Object {
            "releaseDate": "2018-07-07",
            "summary": "This is a fake scroll",
            "title": "There's no hope",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          src="starWarsLogo.jpg"
/>,
        <section
          className="scroll"
>
          <h3>
                    This is a fake scroll
          </h3>
          <h3>
                    There's no hope
          </h3>
          <h3>
                    2018-07-07
          </h3>
</section>,
      ],
      "className": "sidebar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "starWarsLogo.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              This is a fake scroll
</h3>,
            <h3>
              There's no hope
</h3>,
            <h3>
              2018-07-07
</h3>,
          ],
          "className": "scroll",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "This is a fake scroll",
            },
            "ref": null,
            "rendered": "This is a fake scroll",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "There's no hope",
            },
            "ref": null,
            "rendered": "There's no hope",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2018-07-07",
            },
            "ref": null,
            "rendered": "2018-07-07",
            "type": "h3",
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            src="starWarsLogo.jpg"
/>,
          <section
            className="scroll"
>
            <h3>
                        This is a fake scroll
            </h3>
            <h3>
                        There's no hope
            </h3>
            <h3>
                        2018-07-07
            </h3>
</section>,
        ],
        "className": "sidebar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "starWarsLogo.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                This is a fake scroll
</h3>,
              <h3>
                There's no hope
</h3>,
              <h3>
                2018-07-07
</h3>,
            ],
            "className": "scroll",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "This is a fake scroll",
              },
              "ref": null,
              "rendered": "This is a fake scroll",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "There's no hope",
              },
              "ref": null,
              "rendered": "There's no hope",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2018-07-07",
              },
              "ref": null,
              "rendered": "2018-07-07",
              "type": "h3",
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
